pkg/app/pipectl/cmd/appconfig/migratefromdeployconfig.go

```diff
@@ -0,0 +1,186 @@
+	cmd.MarkFlagRequired("repo-id")
+	cmd.MarkFlagRequired("env-name")
+
+	return cmd
+}
+
+func (m *migrateFromDeployConfig) run(ctx context.Context, _ cli.Input) error {
+	cli, err := m.root.clientOptions.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to initialize client: %w", err)
+	}
+	defer cli.Close()
+
+	fmt.Fprintln(m.stdout, "Start finding and migrating deployment configuration files to application configuration files...")
+
+	var cursor string
+	var appConfigs = make(map[string][]byte, 0)
+	for {
+		req := &apiservice.ListApplicationsRequest{
+			EnvName: m.envName,
+			Cursor:  cursor,
+		}
+		resp, err := cli.ListApplications(ctx, req)
+		if err != nil {
+			return fmt.Errorf("failed to list application: %w", err)
+		}
+
+		for _, app := range resp.Applications {
+			if app.GitPath.Repo.Id != m.repoID {
+				continue
+			}
+
+			path := filepath.Join(m.repoRootPath, app.GitPath.GetApplicationConfigFilePath())
+			oriData, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			newData, err := convert(oriData, app.Name, m.envName, app.Description)
+			if err != nil {
+				return err
+			}
+			appConfigs[path] = newData
+		}
+		if resp.Cursor == "" {
+			break
+		}
+		cursor = resp.Cursor
+	}
+
+	if len(appConfigs) == 0 {
+		fmt.Println("There is no applications to migrate")
+		return nil
+	}
+
+	for p, data := range appConfigs {
+		info, err := os.Stat(p)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(p, data, info.Mode()); err != nil {
+			return err
+		}
+	}
+
+	fmt.Fprintf(m.stdout, "Successfully migrated %d applications\n", len(appConfigs))
+	return nil
+}
+
+func convert(data []byte, name, env, description string) ([]byte, error) {
+	var out strings.Builder
+	var shouldWrite, done bool
+
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+	for scanner.Scan() {
+		line := scanner.Text()
+		if shouldWrite {
+			indent, ok := determineIndent(line)
+			if !ok {
+				fmt.Fprintf(&out, "%s\n", line)
+				continue
+			}
+
+			writeNewFields(&out, name, env, description, indent)
+			shouldWrite = false
+			done = true
+		}
+
+		fmt.Fprintf(&out, "%s\n", line)
+		if done {
+			continue
+		}
+		if strings.HasPrefix(line, "spec:") {
+			shouldWrite = true
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+	return []byte(out.String()), nil
+}
+
+func determineIndent(line string) (string, bool) {
+	noIndentLine := strings.TrimLeft(line, " \t")
+	// In case of just an empty line.
+	if noIndentLine == "" {
+		return "", false
+	}
+	// In case of just a comment.
+	if strings.HasPrefix(noIndentLine, "#") {
+		return "", false
+	}
+	return line[:len(line)-len(noIndentLine)], true
+}
+
+func writeNewFields(out io.Writer, name, env, description, indent string) {
+	doubleIndent := strings.Repeat(indent, 2)
+
+	fmt.Fprintf(out, "%sname: %s\n", indent, name)
+	fmt.Fprintf(out, "%slabels:\n", indent)
+	fmt.Fprintf(out, "%senv: %s\n", doubleIndent, env)
```

Comment:
I'm a bit worried about this since the `name` setting via application configuration is released from `v0.23.0` already, someone may use application configuration files with that field already; thus this write is duplicated. But we may ignore that since users have manual steps after running this command while making PR to update their remote git. If everyone is aware of that already, it's okay to me 👍

Reply:
I that so too. 👍

Reply:
in that case fine to modify manually 👍